students/mark/Session04/file_read_parse_lab.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+
+"""
+File reading and parsing
+
+Download this text file:
+
+students.txt
+
+In it, you will find a list of names and what programming languages they have used in the past.
+This may be similar to a list generated at the beginning of this class.
+
+Write a little script that reads that file, and generates a list of all the languages that have been used.
+
+What might be the best data structure to use to keep track of bunch of values without duplication?
+"""
+
+
+### Note: we could assume we know all the languages, however, we're instead going to assumet everyone capitalizes thier nickname.
+valid_language=(
+    "ansible",
+    "bash",
+    "c",
+    "c#",
+    "c++",
+    "db",
+    "erlang",
+    "fortran",
+    "java",
+    "javascript",
+    "lisp",
+    "matlab",
+    "mysql",
+    "nothing",
+    "perl",
+    "php",
+    "powershell",
+    "python",
+    "r",
+    "rex",
+    "ruby",
+    "shell",
+    "sql",
+    "typescript",
+    "vb",
+    "visualbasic",
+    )
+
+
+students_file="./students.txt"
+
+with open(students_file, 'r') as f:
+    read_data = f.readlines()
+f.closed
+
```

Comment:
if you're using "with open()" you don't need to close the file afterwards

Reply:
Got it, thanks, removed f.closed line.